src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -800,14 +800,39 @@ private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer
             var usePowerFxV1CompatibilityRules = features.PowerFxV1CompatibilityRules;
 
             // Special cases for comparing option set values
-            // Without StronglyTypedBuiltinEnums, option sets are convereted to their backing kind before getting here in Visitor.Visit(FirstNameNode)
+            // Without StronglyTypedBuiltinEnums, enums are convereted to their backing kind before getting here in Binder.cs, Visitor.Visit(FirstNameNode)
             if (typeLeft.Kind == DKind.OptionSetValue || typeRight.Kind == DKind.OptionSetValue)
             {
-                // Comparing values from two different option set values is not supported
-                if (typeLeft.Kind == DKind.OptionSetValue && typeRight.Kind == DKind.OptionSetValue
-                    && typeLeft.Accepts(typeRight, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules))
+                if (usePowerFxV1CompatibilityRules)
                 {
-                    if (!usePowerFxV1CompatibilityRules || typeLeft.OptionSetInfo.CanCompareNumeric)
+                    // Comparing values from two different option set values is not supported
+                    if (typeLeft.Kind == DKind.OptionSetValue && typeRight.Kind == DKind.OptionSetValue
+                        && typeLeft.Accepts(typeRight, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules))
+                    {
+                        if (!usePowerFxV1CompatibilityRules || typeLeft.OptionSetInfo.CanCompareNumeric)
```

Comment:
>!usePowerFxV1CompatibilityRules ||  [](http://example.com/codeflow?start=28&length=35)

Can be removed, will always be false (because of `if` on 806)

Reply:
I'll an issue to resolve this later.